### PR TITLE
Support for maven and gradle wrappers

### PR DESCRIPTION
**Pull request by fabio-franco**
This task adds the features for Gradle and Maven settings for Intellij #39 . 

## Maven

image

allows end-users specify:
- Bundled (use default mvn)
- Maven wrapper (use mvnw)
- "..." -> Custom maven

## Gradle

image

allows end-users specify:
- "wrappers" -> gradle-wrapper / wrapper
- "Specific location" -> Custom/Default Gradle

## Tests

- All tests so far were done on Mac. It's necessary to cover Windows and Linux environments!

### CODE REVIEW

**Comment by fabio-franco** [comment]
I just tested `mvnw` on my Windows VM and it seems that for windows it's necessary to do `.\mvnw` to execute maven wrapper:
image

File: src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java [hunk 1]
```diff
@@ -153,4 +158,43 @@ private static boolean containerVersion(String plugin) {
...
+                if (gradlePath != null) {
+                    return gradlePath;
+                }
+            }
+        }
+        return "gradle"; // default gradle
+    }
+
+    private static String getLocalGradleWrapperPath(Project project) {
+        String gradlew = SystemInfo.isWindows ? "gradlew.bat" : "./gradlew";
```
Comment (evie-lau) [inline src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java:185]: Make sure to add the `./` for Windows' gradlew.bat. 

Ran into this issue testing: 
image

Comment (fabio-franco) [reply src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java:185]: Do you mean `.\`, right? Or `./` also works on Windows?

**Comment by fabio-franco** [comment]
On my last commit, I included some fixes for Windows, including `.\` and `cmd /K` before the Gradle and Maven commands, just for Windows. Without it, during execution, a new terminal window was opened. The reasons seem related to the difference between the terminal console runtime level and the `gradlew.bat` and `maven.cmd` level (Console application vs Windows application). https://superuser.com/questions/244392/control-whether-windows-command-line-commands-run-in-new-window

**Review by evie-lau — APPROVED** [review]
LGTM. 
I tested on my Windows environment and it seems all the different options work as intended. (I did rebase the branch on top of my existing java17 branch so I could use `runIde` with fewer problems, but I didn't change any of this new code)

File: src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java [hunk 2]
```diff
@@ -153,4 +158,44 @@ private static boolean containerVersion(String plugin) {
...
+                    return gradlePath;
+                }
+            }
+        }
+        return "gradle"; // default gradle
+    }
+
+    private static String getLocalGradleWrapperPath(Project project) {
+        String gradlew = SystemInfo.isWindows ? ".\\gradlew.bat" : "./gradlew";
+        File file = new File(project.getBasePath() + File.separator + gradlew);
```
Comment (turkeylurkey) [inline src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java:186]: Suggestion, we usually use the File(File, String) constructor in a case like this.

File: src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java [hunk 3]
```diff
@@ -153,4 +158,44 @@ private static boolean containerVersion(String plugin) {
...
+        return file.exists() ? gradlew : null;
+    }
+
+    private static String getCustomGradlePath (String customGradleHome) {
+        String additionalCMD = SystemInfo.isWindows ? "cmd /K " : ""; // without it, a new terminal window is opened
+        File gradleHomeFile = new File(customGradleHome);
+        if (gradleHomeFile != null) {
+            // When a custom gradle is specified, IntelliJ settings force it to point to the root folder and consider the subfolders invalid,
+            // and consequently, it will return null. For this reason, we need to use ./bin/gradle in order to execute gradle.
+            File file = new File(gradleHomeFile.getAbsolutePath() + File.separator + "bin"+ File.separator + "gradle");
```
Comment (turkeylurkey) [inline src/main/java/io/openliberty/tools/intellij/util/LibertyGradleUtil.java:196]: Again, the File(File, String) constructor would be simpler.

**Review by turkeylurkey — APPROVED** [review]

**Comment by fabio-franco** [comment]
@turkeylurkey  After the conference today I will submit the proposed changes 😀

**Review by turkeylurkey — APPROVED** [review]

**Pull request merged**